cmake/ImpactXFunctions.cmake

```diff
@@ -124,15 +124,23 @@ macro(impactx_set_default_install_dirs_python)
 endmacro()
 
 
-# function to set the PYTHONPATH on a test
+# function to set the PYTHONPATH and PATH (for .dll files) on a test
 # this avoids that we need to install our python packages to run ctest
 #
 function(impactx_test_set_pythonpath test_name)
     if(WIN32)
         string(REPLACE ";" "\\;" WIN_PYTHONPATH "$ENV{PYTHONPATH}")
+        string(REPLACE ";" "\\;" WIN_PATH "$ENV{PATH}")  # DLLs
         string(REGEX REPLACE "/" "\\\\" WIN_PYTHON_OUTPUT_DIRECTORY ${CMAKE_PYTHON_OUTPUT_DIRECTORY})
+        # shared library note:
+        #   For Windows Python 3.8+, this also needs to be injected via
+        #   os.add_dll_directory.
+        #   https://github.com/python/cpython/issues/80266
+        #   https://docs.python.org/3.8/library/os.html#os.add_dll_directory
         set_property(TEST ${test_name}
-            APPEND PROPERTY ENVIRONMENT "PYTHONPATH=${WIN_PYTHON_OUTPUT_DIRECTORY}\;${WIN_PYTHONPATH}"
+            APPEND PROPERTY ENVIRONMENT
+                "PYTHONPATH=${WIN_PYTHON_OUTPUT_DIRECTORY}\;${WIN_PYTHONPATH}"
+                "PATH=$<TARGET_FILE_DIR:lib>\;${WIN_PATH}$"
```

Comment:
while `.lib` is the build-time location of `libimpactx.dll`, we should also inject here the build-time location of it's dependent `.dll`s, including `libamrex.dll`.